contrib/fuzz/containerd_import_fuzz_test.go

```diff
@@ -31,30 +31,35 @@ func fuzzContext() (context.Context, context.CancelFunc) {
 	return ctx, cancel
 }
 
-func FuzzContainerdImport(data []byte) int {
-	initDaemon.Do(startDaemon)
-
-	client, err := containerd.New(defaultAddress)
-	if err != nil {
-		return 0
+func FuzzContainerdImport(f *testing.F) {
+	if os.Getuid() != 0 {
+		f.Skip("skipping fuzz test that requires root")
 	}
-	defer client.Close()
 
-	f := fuzz.NewConsumer(data)
+	f.Fuzz(func(t *testing.T, data []byte) {
+		initDaemon.Do(startDaemon)
 
-	noOfImports, err := f.GetInt()
-	if err != nil {
-		return 0
-	}
-	maxImports := 20
-	ctx, cancel := fuzzContext()
-	defer cancel()
-	for i := 0; i < noOfImports%maxImports; i++ {
-		tarBytes, err := f.GetBytes()
+		client, err := containerd.New(defaultAddress)
 		if err != nil {
-			return 0
+			return
```

Comment:
Don't you need to call Fatalf on the non-nil error cases?

Reply:
no, this means the data generation from `fuzz.NewConsumer` is insufficient and and we can return directly to let fuzzer retry another input.

basically, if previous fuzz uses `return 0/1`, we can `return` in go native fuzz directly; if previous fuzz `panic(err)`, we should `t.Fatal(err)`.

similar to https://github.com/moby/moby/pull/44458/files#r1027076666

Reply:
Sorry. The lines were confusing.

containerd.New() doesn't take a random input. We can panic on the error. Although, we didn't do that in the original code though.

client.Import() takes a random input. We can ignore the error as you said.

Reply:
ah you're right, good finding. Updated both tests to `t.Fatal` on `containerd.New()`.